File: test/front-end/src/app/service/export-bill/exportbill.service.ts
import { Injectable } from '@angular/core';
import {HttpClient} from '@angular/common/http';
import {Observable} from 'rxjs';
import {ExportBill} from '../../model/export-bill/exportBill';
import {ExportBillDetails} from '../../model/export-bill/exportBillDetails';

@Injectable({
  providedIn: 'root'
})
export class ExportbillService {
  private API_EXPORT_BILL = "http://localhost:8080/api/export-bill";
  private API_EMPLOYEE = "http://localhost:8080/api";
  constructor(private http: HttpClient) { }

  // @ts-ignore
  createCodeExportBillRefund(): Observable<any> {
    return this.http.get(`${this.API_EXPORT_BILL}/createCode`);
  }

  createExportBill(exportBill: ExportBill): Observable<any>{
    return this.http.post<any>(this.API_EXPORT_BILL,exportBill);
  }

  createExportBillDetail(exportBillDetail : ExportBillDetails): Observable<any>{
    return this.http.post<any>(this.API_EXPORT_BILL+ '/export-bill-detail',exportBillDetail);
  }

  getEmployee():Observable<any>{
    return this.http.get<String>(this.API_EMPLOYEE+ '/employees');
  }

  getAllCaseRecord(pageNumber: number): Observable<ExportBill[]> {
    return this.http.get<ExportBill[]>(this.API_EXPORT_BILL + '?page=' + pageNumber);
  }

  getCaseRecordById(id: number): Observable<ExportBill> {
    return this.http.get<ExportBill>(this.API_EXPORT_BILL + '/find/' + id);
  }

  // tslint:disable-next-line:max-line-length
  getCaseRecordByFields(field1: string, field2: string, field3: string, field4: string, field5: string, pageNumber: number): Observable<ExportBill[]> {
    // tslint:disable-next-line:max-line-length
    return this.http.get<ExportBill[]>(this.API_EXPORT_BILL + '/find/' + field1 + '/' + field2 + '/' + field3 + '/' + field4 + '/' + field5 + '?page=' + pageNumber);
  }

  delete(id: number): void {
    this.http.delete(this.API_EXPORT_BILL + '/delete/' + id );
  }

}

File: test/back-end/src/main/java/com/example/demo/model/service/ExportBillService.java
package com.example.demo.model.service;

import com.example.demo.model.entity.export_bill.ExportBill;
import com.example.demo.model.entity.export_bill.ExportBillDetail;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.util.List;

public interface ExportBillService {
//    void createExportBill(ExportBill exportBill);
//    void createExportBillDetail(ExportBillDetail exportBillDetail);
//    String createExportBillRefundCode();
//    List<String> getAllExportBillCode();

    //List
    Page<ExportBill> findAll(Pageable pageable);
    ExportBill findById(long id);
    Page<ExportBill> findByFields(String field1, String field2, String field3, String field4, Long typeId, Pageable pageable);
    void delete(long id);
}
